src/layout/hierarchy/index.ts

```diff
@@ -79,6 +79,31 @@ const _hierarchyToGraph = (hierarchy: HierarchyPointNode<Hierarchy>, nodesById:
 
 const hierarchyToGraph = (hierarchy: HierarchyPointNode<Hierarchy>) => _hierarchyToGraph(hierarchy, {})
 
+const findSubgraphNode = (node: Node, id: string): Node | undefined => {
+  if (node.id === id) return node
+
+  if (node.subgraph !== undefined) {
+    for (const subgraphNode of node.subgraph.nodes) {
+      const nestedNode = findSubgraphNode(subgraphNode, id)
+      if (nestedNode !== undefined) return nestedNode
+    }
+  }
+
+  return undefined
+}
+
+const findParent = (nodes: Node[], id: string): Node | undefined => {
```

Comment:
can we rewrite this so that it returns the top level node if that node matches id? And returns the node id, rather than node? I think the current implementation will return undefined if you pass it an id of a top-level node (i.e. it should consider a top level node it's own ancestor). probably means we should rename to something like `findAncestor`. the implementation would be nearly the same:

```ts
for (const node of nodes) {
  if (node.id === id) {
    return node.id
  }

  for (const subgraphNode of node.subgraph?.nodes ?? []) {
    const nestedNode = findSubgraphNode(subgraphNode, id)
    if (nestedNode !== undefined) return node.id
  }
}
```